src/libAtomVM/posix_nifs.c

```diff
@@ -0,0 +1,420 @@
+    // These are defined in SUSv1
+    switch (err) {
+        case EACCES:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "eacces"));
+        case EAGAIN:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "eagain"));
+        case EBADF:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "ebadf"));
+        case EBUSY:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "ebusy"));
+        case EDQUOT:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "edquot"));
+        case EEXIST:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "eexist"));
+        case EFAULT:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "efault"));
+        case EFBIG:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "efbig"));
+        case EINTR:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "eintr"));
+        case EINVAL:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "einval"));
+        case EIO:
+            return globalcontext_make_atom(glb, ATOM_STR("\x3", "eio"));
+        case EISDIR:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "eisdir"));
+        case ELOOP:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "eloop"));
+        case EMFILE:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "emfile"));
+        case EMLINK:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "emlink"));
+        case ENAMETOOLONG:
+            return globalcontext_make_atom(glb, ATOM_STR("\xC", "enametoolong"));
+        case ENFILE:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "enfile"));
+        case ENODEV:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "enodev"));
+        case ENOENT:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "enoent"));
+        case ENOMEM:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "enomem"));
+        case ENOSPC:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "enospc"));
+        case ENOTDIR:
+            return globalcontext_make_atom(glb, ATOM_STR("\x7", "enotdir"));
+        case ENXIO:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "enxio"));
+        case EPERM:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "eperm"));
+        case EPIPE:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "epipe"));
+        case EROFS:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "erofs"));
+        case ESPIPE:
+            return globalcontext_make_atom(glb, ATOM_STR("\x6", "espipe"));
+        case ESRCH:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "esrch"));
+        case EXDEV:
+            return globalcontext_make_atom(glb, ATOM_STR("\x5", "exdev"));
+    }
+#else
+    UNUSED(err);
+    UNUSED(glb);
+#endif
+    return UNDEFINED_ATOM;
```

Comment:
I was wondering here if instead of `undefined` we should to a term_from int so we at least get the errno in any errors that are returned.

Reply:
Let's rename the function then to `posix_errno_to_term`.